### PR TITLE
json_serializable: refactor enum logic

### PR DESCRIPTION
**Pull request by kevmoo**
Refactor a function into to functions
Simplify the return value

### CODE REVIEW

**Comment by kevmoo** [comment]
FYI @natebosch – you might could disable auto-merge if you think I may want to change things given your feedback.

File: json_serializable/lib/src/enum_utils.dart [hunk 1]
```diff
@@ -27,73 +27,78 @@ String? enumValueMapFromType(
...
+
+  final items = enumMap.entries
+      .map((e) => '  ${targetType.element2!.name}.${e.key.name}: '
+          '${jsonLiteralAsDart(e.value)},')
+      .join();
+
+  return 'const ${constMapName(targetType)} = {\n$items\n};';
+}
+
+Object? generateEntry({
```
Comment (natebosch) [inline json_serializable/lib/src/enum_utils.dart:47]: [nit] can this be private?

**Review by natebosch — APPROVED** [review]

File: json_serializable/lib/src/enum_utils.dart (hunk 1 above)
Comment (natebosch) [reply json_serializable/lib/src/enum_utils.dart:47]: Ugh, I need to remember to check the "request changes" box when I leave comments...

**Comment by natebosch** [comment]
> FYI @natebosch – you might could disable auto-merge if you think I may want to change things given your feedback.

Yeah, I thought about that after the last one got merged and then didn't think about it at all when reviewing this one...

I'll get the reflexes to check that box at some point 😝

**Comment by kevmoo** [comment]
Ha!

**Pull request merged**